### PR TITLE
fix(api): Adding fields name, level and parentId in collection schema

### PR DESCRIPTION
**Pull request by ghost**
## What's the purpose of this pull request?

This PR adds three fields in the 'allCollections' query: name, level and parentId. I need them to build the menu inside my store header, but they were not in the Gatsby data layer. In front, I use this data to build the category tree for the menu.

## How it works? 

Adds name, level and parentId properties in the GraphQL schema;
Adds level and parentId properties in CategoryTree interface;
Updates collection fetching in query resolver to append parentId.

## How to test it?

Check the base.store PR created below and try to run allCollection and collection queries with the new fields.

### `base.store` Deploy Preview

[vtex-sites/base.store#223](https://github.com/vtex-sites/base.store/pull/223)

## References
Followed the tutorial from Office Hours of 06/Jan/2022.

### CODE REVIEW

File: packages/api/src/platforms/vtex/clients/commerce/types/CategoryTree.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 export interface CategoryTree {
   id: number
+  level?: number
```
Comment (icazevedo) [inline packages/api/src/platforms/vtex/clients/commerce/types/CategoryTree.ts:3]:
```suggestion
```

File: packages/api/src/typeDefs/collection.graphql [hunk 2]
```diff
@@ -21,6 +21,9 @@ type StoreCollection {
   breadcrumbList: StoreBreadcrumbList!
   meta: StoreCollectionMeta!
   id: ID!
+  name: String!
   slug: String!
   type: StoreCollectionType!
+  level: Int
```
Comment (icazevedo) [inline packages/api/src/typeDefs/collection.graphql:27]: According to the return on the function above, I think we can make it not null

```suggestion
  level: Int!
```

File: packages/api/src/typeDefs/collection.graphql [hunk 3]
```diff
@@ -21,6 +21,9 @@ type StoreCollection {
...
+  parentId: String
```
Comment (icazevedo) [inline packages/api/src/typeDefs/collection.graphql:28]: Is it a number or a string? Not clear from the way you typed on TS and here on the schema

**Review by icazevedo — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: packages/api/src/typeDefs/collection.graphql (hunk 3 above)
Comment (brunightplus) [reply packages/api/src/typeDefs/collection.graphql:28]: It should be the same of the `id` field, I think. In TS, `id` is a number, but in the schema it returns a string, so I set `parentId` as a string there too

**Pull request closed without merging**